libraries/AP_Scripting/examples/plane-doublets.lua

```diff
@@ -6,15 +6,15 @@
 -- Magnitude and duration of the doublet can also be controlled.
 -- It is suggested to allow the aircraft to trim for straight, level, unaccelerated flight (SLUF) in FBWB mode before
 -- starting a doublet
--- Charlie Johnson, Oklahoma State University 2020
+-- Charlie Johnson, Oklahoma State University 2020, modification by Astik Srivastava, Delhi Technological University 2024
 
 local DOUBLET_ACTION_CHANNEL = 6 -- RCIN channel to start a doublet when high (>1700)
 local DOUBLET_CHOICE_CHANNEL = 7 -- RCIN channel to choose elevator (low) or rudder (high)
-local DOUBLET_FUCNTION = 19 -- which control surface (SERVOx_FUNCTION) number will have a doublet happen
+local DOUBLET_FUCNTION = 1 -- which control surface (SERVOx_FUNCTION) number will have a doublet happen
 -- A (Servo 1, Function 4), E (Servo 2, Function 19), and R (Servo 4, Function 21)
 local DOUBLET_MAGNITUDE = 6 -- defined out of 45 deg used for set_output_scaled
-local DOUBLET_TIME = 500 -- period of doublet signal in ms
-
+local DOUBLET_TIME = 2000 -- period of doublet signal in ms
```

Comment:
This period seems rather large. I guess it depends on the mode being excited by the doublet and the aircraft category of performance. For sporty planes the 500 ms (2 Hz) period might even be a bit long.

This doublet input is most useful for the fast second order modes of short period and Dutch roll. Not very helpful for phugoid, roll, or spiral.

It might be helpful to pull in a table of suggested values for each category and mode. I believe the Nelson textbook has one.